### PR TITLE
Bump OTel 1.39. and instrumentation to 2.5.0

### PR DESCRIPTION
**Pull request by brunobat**
**BREAKING CHANGES.**

This updates the opentelemetry-java bom from 1.32 to 1.39 and the opentelemetry-instrumentation alpha bom from 1.32 to 2.50.

The breaking changes for the users are:

- HTTP semantic conventions have changed. See the full [list of changes](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/migration-guide.md#summary-of-changes);
- `quarkus.otel.semconv-stability.opt-in` system property was removed because users cannot opt-in anymore;
- DB Span names have changed in the case of table creation;
- Deprecated annotation for `io.opentelemetry.extension.annotations.WithSpan` has been removed. Please use the new `io.opentelemetry.instrumentation.annotations.WithSpan`, as previously announced.

For later: 

- The default OTLP protocol has been changed from `grpc` to `http/protobuf` in order to align with the [specification](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.28.0/specification/protocol/exporter.md#specify-protocol). See https://github.com/quarkusio/quarkus/issues/41522 

Other things to look at:

- Removed obsolete tests for the now removed `@WithSpan` legacy annotation.
- Removed the triple test executions depending on the value of the `quarkus.otel.semconv-stability.opt-in` system property. This will speed up tests considerably. 
- Documentation updates.

For more details please check:
https://github.com/open-telemetry/opentelemetry-java/blob/main/CHANGELOG.md
https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/CHANGELOG.md

### CODE REVIEW

**Comment by brunobat** [comment]
Will squash the commits after the PR is reviews as it makes it easy to spot change implications.

**Comment by brunobat** [comment]
This will depend on SmallRye reactive messaging changes being currently made... Will post PR here, when available.

**Review by radcortez — APPROVED** [review]
At least we got rid of a bunch of code :)

File: extensions/opentelemetry/runtime/pom.xml [hunk 1]
```diff
@@ -104,7 +100,7 @@
         </dependency>
         <dependency>
             <groupId>io.opentelemetry</groupId>
-            <artifactId>opentelemetry-api-events</artifactId>
+            <artifactId>opentelemetry-api-incubator</artifactId>
```
Comment (trask) [inline extensions/opentelemetry/runtime/pom.xml:103]: what do you use this module for?

File: extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/cdi/WithSpanInterceptor.java [hunk 2]
```diff
@@ -22,11 +22,11 @@
 import io.opentelemetry.instrumentation.annotations.WithSpan;
 import io.opentelemetry.instrumentation.api.annotation.support.MethodSpanAttributesExtractor;
 import io.opentelemetry.instrumentation.api.annotation.support.ParameterAttributeNamesExtractor;
+import io.opentelemetry.instrumentation.api.incubator.semconv.util.SpanNames;
```
Comment (trask) [inline extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/cdi/WithSpanInterceptor.java:25]: can you open an issue about stabilizing this class?

**Review by trask — COMMENTED** [review]
❤️ sorry you have had to deal with all of the changes in these unstable components. it looks like most of the remaining unstable dependencies are related to DB, RPC and messaging semconv. I think DB and messaging semconv will be stable by end of the year, and we will start working on RPC semconv stability in the next few months. In the opentelemetry-java-instrumentation repo we are currently planning to bump from 2.x to 3.x once we are ready to switch over to stable semconv for all three (DB, RPC and messaging) by default. I noticed that you followed the opt-in flag for http which is awesome, but I would definitely understand if you didn't go through that for future updates since it's quite painful unless you are getting pushback on this from your users (in general we haven't had too much expectation of libraries emitting native instrumentation before the related semconv is stabilized).

File: extensions/opentelemetry/runtime/pom.xml (hunk 1 above)
Comment (brunobat) [reply extensions/opentelemetry/runtime/pom.xml:103]: We need to reset for tests the `GlobalEventLoggerProvider` otherwhise it throws an error when we execute dev mode. I remember that native would also failed without it. 
It's usage comes from the SDK Autoconfiguration.

File: extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/cdi/WithSpanInterceptor.java (hunk 2 above)
Comment (brunobat) [reply extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/cdi/WithSpanInterceptor.java:25]: Sure! https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/11693

**Comment by gsmet** [comment]
@brunobat I let you squash and merge when you're ready.

**Comment by brunobat** [comment]
Depends on https://github.com/smallrye/smallrye-reactive-messaging/pull/2677

**Review by alesj — APPROVED** [review]

File: tcks/pom.xml [hunk 3]
```diff
@@ -116,7 +116,7 @@
                 <module>microprofile-rest-client</module>
                 <module>microprofile-rest-client-reactive</module>
                 <module>microprofile-openapi</module>
-                <module>microprofile-opentelemetry</module>
+<!--                <module>microprofile-opentelemetry</module>-->
```
[the remark this answers is not in the document]
Comment (brunobat) [reply tcks/pom.xml:119]: yes, the TCK fails because it's using outdated semantic conventions

**Review by ozangunalp — APPROVED** [review]
Needs rebasing for version change

**Pull request merged**